addons/ondevice-controls/src/ControlsPanel.tsx

```diff
@@ -52,19 +52,29 @@ const ControlsPanel = ({ api }: { api: API }) => {
   const store = api.store();
   const storyId = store.getSelection().storyId;
   const [isPristine, setIsPristine] = useState(true);
-  const [argsfromHook, updateArgs, resetArgs] = useArgs(storyId, store);
-  const { argTypes, parameters } = store.fromId(storyId) as StoryContext;
-
-  const argsObject = Object.entries(argTypes).reduce((prev, [key, argType]) => {
-    const isControl = Boolean(argType?.control);
-
-    return isControl
-      ? {
-          ...prev,
-          [key]: { ...argType, name: key, type: argType?.control?.type, value: argsfromHook[key] },
-        }
-      : prev;
-  }, {});
+  const [argsFromHook, updateArgs, resetArgs] = useArgs(storyId, store);
+  const { argsObject, argTypes, parameters } = React.useMemo(() => {
+    const { argTypes, parameters } = store.fromId(storyId) as StoryContext;
```

Comment:
the variable name is already being used so I'm getting eslint warnings here

Reply:
I renamed all the internal variables to avoid the warning about name shadowing.